vm/stardust/vm_stardust.go

```diff
@@ -465,13 +466,12 @@ func accountBlockIssuerSTVF(input *vm.ChainOutputWithIDs, currentBlockIssuerFeat
 
 	// AccountInPotential
 	// the storage deposit does not generate potential mana, so we only use the excess base tokens to calculate the potential mana
-	var excessBaseTokensAccount iotago.BaseToken
-	minDeposit := rentStructure.MinDeposit(current)
-	if current.BaseTokenAmount() <= minDeposit {
-		excessBaseTokensAccount = 0
-	} else {
-		excessBaseTokensAccount = current.BaseTokenAmount() - minDeposit
+	minDeposit, err := rentStructure.MinDeposit(current)
+	if err != nil {
+		return err
 	}
+	// no need to check underflow error as the result of safe sub will be zero in that case
+	excessBaseTokensAccount, _ := safemath.SafeSub(current.BaseTokenAmount(), minDeposit)
```

Comment:
Better check for the error here and replace the excessBaseTokensAccount with 0 if you get the underflow error.